HACKING.md

```diff
@@ -33,7 +33,9 @@ Edit the `./environments/development.env` file and adjust the configurations var
 | `OCAML_BENCH_DB_PASSWORD` | `docker`      |
 | `OCAML_BENCH_GRAPHQL_KEY` | `secret`      |
 | `OCAML_BENCH_TARGET_ARCH` | `amd64`       |
-
+| `VITE_GRAPHQL_URL` | `http://localhost:8080/v1/graphql ` |
+| `VITE_PIPELINE_FRONTEND_URL` | `http://localhost:8080 ` |
+| `VITE_PIPELINE_PORT` | `8081` |
```

Comment:
I disagree with the naming for the following reasons:

1. Vite is just a tool, we shouldn't leak it into global config names.
  - If we decide to replace it by something else, we shouldn't worry about replacing these names.
  - People who want to run our services might not be familiar with vite and this will require them to understand what it is. In the end I think it will just confuse them ("Is this some kind of configuration for the JS build system?").
  - Vite is *not* actually used in production. We serve the built production website with nginx, thus using `VITE_PIPELINE_FRONTEND_URL` in production does not make sense.
2. `VITE_GRAPHQL_URL` is wrong: vite has nothing to do with GraphQL in our setup. I suggest we use the same name here as we use for the service in the `docker-compose` file – `GRAPHQL_ENGINE`.
3. The names are missing the `OCAML_BENCH_` prefix. I added it intentionally to have consistency for all our env vars and to avoid conflicts with any other env vars users might already have.

Below is my suggestion for service names. By service I mean something that listens on a network port.

| Service ID | Development port | Production port | Description |
|--|--|--|--|
| `OCAML_BENCH_FRONTEND` | 8082 | 80 | The main application frontend service for benchmark visualisation as implemented in the `./frontend` folder. This is served by vite in development mode and by nginx in production. |
| `OCAML_BENCH_PIPELINE_UI` | 8081 | 8081 | The pipeline UI as implemented by OCurrent. |
| `OCAML_BENCH_GRAPHQL_ENGINE` | 8080 | 8080 | The pipeline UI as implemented by OCurrent. |
| `OCAML_BENCH_DB` | 5432 | 5432 (internal) | The PostgreSQL database port. In production the service is only available internally, i.e, it is not exposed to the internet. |


If this makes sense, can we make the naming changes for `_URL` and `_PORT` to be consistent with the above?

Reply:
If we do not split them up, we'll have to "hardcode" the port numbers in the `docker-compose.yaml` file. It means that if someone wants to change one of the service's port, he will have to make the change at two places. Probably not a big deal but still wanted to raise the point :)